assets/src/edit-story/components/helpCenter/useHelpCenter.js

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { useReducer, useEffect, useState, useMemo } from 'react';
+/**
+ * Internal dependencies
+ */
+import { clamp } from '../../../animation';
+import { NAVIGATION_FLOW } from './constants';
+
+/**
+ * Performs any state updates that result from
+ * inspection of previous or next state
+ *
+ * @param {*} previous - previous state
+ * @param {*} next - next state
+ * @return {*} partial of state
+ */
+export const deriveState = (previous, next) => {
+  const isMenuIndex = next.navigationIndex < 0;
+  const isLTRTransition = next.navigationIndex - previous.navigationIndex > 0;
+
+  return {
+    hasBottomNavigation: !isMenuIndex,
+    isLeftToRightTransition: !isMenuIndex && isLTRTransition,
+    isPrevDisabled: next.navigationIndex <= 0,
+    isNextDisabled: next.navigationIndex >= next.navigationFlow?.length - 1,
+  };
+};
+
+const reducer = ({ state, actions }, action) => {
+  const nextState = { ...state, ...action(state) };
+  return {
+    state: { ...nextState, ...deriveState(state, nextState) },
+    actions,
+  };
+};
+
+const initial = {
+  state: {
+    isOpen: false,
+    navigationIndex: -1,
+    // @TODO make this dynamic based off of unread tips.
+    navigationFlow: NAVIGATION_FLOW,
+    isLeftToRightTransition: true,
+    hasBottomNavigation: false,
+    isPrevDisabled: true,
+    isNextDisabled: false,
+  },
+  // All actions are in the form: externalArgs -> state -> newStatePartial
+  //
+  // Actions should only update the part of state they directly effect.
+  // If there are state updates from deriving a value off of prevState
+  // or nextState, place them in `deriveState(previous, next)` above.
+  actions: {
```

Comment:
maaaaaan it would be so dope to have this all in typescript

Reply:
haah yea would be real nice